geoutils/georaster.py

```diff
@@ -579,6 +580,23 @@ def reproject(self, dst_ref=None, dst_crs=None, dst_size=None, dst_bounds=None,
             dst_data = np.ones(dst_shape)
             reproj_kwargs.update({'destination': dst_data})
 
+        # Check that reprojection is actually needed
+        if all([
+                (dst_transform == self.transform) or (dst_transform is None),
+                (dst_crs == self.crs) or (dst_crs is None),
+                (dst_size == self.shape) or (dst_size is None),
+                (dst_res == self.res) or (dst_res == self.res[0] == self.res[1]) or (dst_res is None)
+        ]):
+            if (nodata == self.nodata) or (nodata is None):
+                warnings.warn("Output projection, bounds and size are identical -> return self")
+                return self
+
+            else:
+                warnings.warn("Only nodata is different, running self.set_ndv instead")
```

Comment:
Wouldn't it be nice to have a keyword argument to disable this warning?

An example where I would get annoyed by the warning:
```python
# I want to make sure that the to-be-aligned DEM has the same shape and extent as the reference
# and I don't care what is done to make sure of this! Maybe they already have the same shape.
# Kind of like np.asarray() just to make sure that it's an array
ref_dem = Raster("path/to/ref")
tba_dem = Raster("path/to/tba").reproject(ref_dem)


```

Reply:
We could possibly have a silent option... Or you can disable warnings...
It's just that in your case, tba_dem and ref_dem are the same objects, so it's a bit dangerous...

Reply:
Your wish was just granted @erikmannerfelt !

Reply:
Nice! 

"path/to/ref" and "path/to/tba" is different so they would not be the same object!

Reply:
Looks like you're right... But then this is not doing what we want, because it means it is still reproject and not returning self...

Reply:
For example:
```
img = gr.Raster(datasets.get_path('landsat_B4'))
img2 = gr.Raster(datasets.get_path('landsat_B4'))
img2_proj = img2.reproject(img)  # This should return img2
print(img2_proj.data[0,0,0])
img2_proj.data[0,0,0] = 0
print(img2_proj.data[0,0,0])
print(img2.data[0,0,0])
```
returns
255
0
255
so it means img2_proj and img2 are different objects...

Reply:
@erikmannerfelt, with commit 08783b1, this is different now. If your dst_ref has the same projection/dimension as your raster, it will just return self, so NOT a copy. Something to be aware of, hence the warning... but at least you can silent it now!